backend/src/services/update-user/update-user.service.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from '@data/prisma-client';
+import { bcryptHash } from '@helpers/crypto/crypto';
+import { Prisma, User } from '@prisma/client';
+
+const updateUser = async (
+  user: Prisma.UserCreateInput,
+  new_password?: string,
```

Comment:
It would be better to pass the `new_password` as part of the `user` object. Because it's actually its part. Then, you can exclude it from the `userData` as const { id, password, ...userData } = user`